textext/base.py

```diff
@@ -305,8 +305,13 @@ def preview_callback(_text, _preamble, _preview_callback, _tex_command, _white_b
                     self.config.save()
 
             else:
-                # ToDo: I think this is completely broken...
-                self.do_convert(self.options.text,
+                # In case TT has been called with --text="" the old node is
+                # just re-compiled if one exists
+                if self.options.text == "" and text is not None:
+                    new_text = text
+                else:
+                    new_text = self.options.text
+                self.do_convert(new_text,
```

Comment:
I would prefer cryptic magic value to new option (e.g. `--recompile`), but it's not a public interface of the plugin and serves for tests only, so I guess it's not worth extra efforts.

Reply:
I agree.